src/dmsort.rs

```diff
@@ -217,13 +217,14 @@ impl<'a, T> Drop for DmSorter<'a, T> {
 unsafe fn unsafe_push<T>(vec: &mut Vec<T>, value: &T) {
 	let old_len = vec.len();
 	vec.reserve(1);
-	ptr::copy_nonoverlapping(value, vec.get_unchecked_mut(old_len), 1);
+	ptr::copy_nonoverlapping(value, vec.as_mut_ptr().add(old_len), 1);
 	vec.set_len(old_len + 1);
 }
 
 #[inline(always)]
 unsafe fn unsafe_copy<T>(slice: &mut [T], source: usize, dest: usize) {
-	ptr::copy_nonoverlapping(slice.get_unchecked(source), slice.get_unchecked_mut(dest), 1);
+	let ptr = slice.as_mut_ptr();
+	ptr::copy_nonoverlapping(ptr.add(source), ptr.add(dest), 1);
```

Comment:
This change is only required with `-Zmiri-tag-raw-pointers`. But also, this could be implemented with `slice::swap_unchecked` if that were stabilized.